net/sscep/Makefile

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (C) 2006 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=sscep
+PKG_VERSION:=0.10.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=v${PKG_VERSION}.tar.gz
```

Comment:
This isn't good, because in dl folder, the tarball will be downloaded as v0.10.0 and it would be hard to know for which sw it was downloaded. It should be packageName-version.

Reply:
This name is based on the naming used by the upstream package; here's the release page:

https://github.com/certnanny/sscep/releases/tag/v0.10.0

And the tar ball that's linked from it:

https://github.com/certnanny/sscep/archive/refs/tags/v0.10.0.tar.gz

Is there a way that I can download it to a different name as the naming. while mandatory for the download,  is highly confusing in the dl folder?

Reply:
@BKPepe this is the last remaining issue raised in the review, I've taken a look at the [documentation](https://openwrt.org/docs/guide-developer/packages) for the variables in a package Makefile and don't see how I can download the package to a different local name than the remote file name reflected in the `PKG_SOURCE`.

Please let me know if you see a way this can be solved. Thanks again!

Reply:
From the link the file name looks like it will be v0.10.0.tar.gz, but if you click the link the download name is actually sscep-0.10.0.tar.gz.

Reply:
That's very interesting @jefferyto, but I don't understand how it works. The actual URI used for the download is https://github.com/certnanny/sscep/archive/refs/tags/v0.10.0.tar.gz and the browser appears to be renaming the file.

Can you give me a raw uri that works and contains the sscep file name? I tried looking at the network tab in my browser to validate what was happening but couldn't find a URI or understand how the apparent rename is occurring.

Reply:
It's the `Content-Disposition: attachment; filename=sscep-0.10.0.tar.gz` header in the final 200 response, after all the redirections.

You can also use codeload as @1715173329 [suggested](https://github.com/openwrt/packages/pull/20683#discussion_r1148317548).

There are many packages here that download from GitHub, you can check other package makefiles to see how they do it.